recipes/flex/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class FlexConan(ConanFile):
+    name = "flex"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/westes/flex"
+    description = "Flex, the fast lexical analyzer generator"
+    topics = ("conan", "flex", "lex", "lexer", "lexical analyzer generator")
+    license = "BSD-2-Clause"    
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    requires = ("m4/1.4.18",)
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Flex package is not compatible with Windows. Consider using winflexbison instead.")
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        configure_args = ["--disable-nls", "HELP2MAN=/bin/true", "M4=m4"]
+        if self.options.shared:
+            configure_args.extend(["--enable-shared", "--disable-static"])
+        else:
+            configure_args.extend(["--disable-shared", "--enable-static"])
+
+        if self.settings.os == "Linux":
+            # https://github.com/westes/flex/issues/247
+            configure_args.extend(["ac_cv_func_malloc_0_nonnull=yes", "ac_cv_func_realloc_0_nonnull=yes"])
+            # https://github.com/easybuilders/easybuild-easyconfigs/pull/5792
+            configure_args.append("ac_cv_func_reallocarray=no")
+
+        # stage1flex must be built on native arch: https://github.com/westes/flex/issues/78
+        # This requires flex to depend on itself.
+        # conan does not support this (currently), so cross build of flex is not possible atm
+
+        self._autotools.configure(args=configure_args)
+        return self._autotools
+
+
+    def build(self):
+        if tools.cross_building(self.settings, skip_x64_x86=True):
```

Comment:
```suggestion
        if tools.cross_building(self, skip_x64_x86=True):
```

Use the conanfile itself as the first argument. We have changed the signature of this function (not breaking, it still works with `settings`) to take advantage of the new xbuilding approach, when using two profiles the conanfile contains the member attributes `settings` and `settings_build`, so this function don¡t need to guess which are the values for the `build` machine.

Reply:
@jgsogo I tried it in https://github.com/conan-io/conan-center-index/pull/1675#issuecomment-630953970 , but I got this failure:
```
icu/64.2: WARN: Build folder is dirty, removing it: /tmp/c3ipr/pr_1675_2_0_1/.conan/data/icu/64.2/_/_/build/42abb35bedb6dbcf0927ff729cf22f3901465469
icu/64.2: ERROR: Package '42abb35bedb6dbcf0927ff729cf22f3901465469' build failed
icu/64.2: WARN: Build folder /tmp/c3ipr/pr_1675_2_0_1/.conan/data/icu/64.2/_/_/build/42abb35bedb6dbcf0927ff729cf22f3901465469
ERROR: icu/64.2: Error in build() method, line 93
	self.run(self._build_config_cmd, win_bash=tools.os_info.is_windows)
while calling '_build_config_cmd', line 180
	return "../source/runConfigureICU %s" % " ".join(self.build_config_args)
while calling 'build_config_args', line 144
	if tools.cross_building(self, skip_x64_x86=True):
	AttributeError: 'ICUBase' object has no attribute 'get_safe'

```

Reply:
Oh! Maybe related to this https://github.com/conan-io/conan/pull/7032, but need to check locally. Ok, leave it as it is, when we start to build using two profiles we should revisit many recipes to make this kind of changes.

Thanks!